oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/OAuth2AuthorizationException.java

```diff
@@ -30,7 +32,18 @@
 	 * @param error the {@link OAuth2Error OAuth 2.0 Error}
 	 */
 	public OAuth2AuthorizationException(OAuth2Error error) {
-		super(error.toString());
+		this(error, error.toString());
+	}
+
+	/**
+	 * Constructs an {@code OAuth2AuthorizationException} using the provided parameters.
+	 *
+	 * @param error the {@link OAuth2Error OAuth 2.0 Error}
+	 * @param message the exception message
+	 */
+	public OAuth2AuthorizationException(OAuth2Error error, String message) {
```

Comment:
Added constructors that take a message, because I wanted the ability to explicitly specify a message that provides more context, without changing the message that was returned in a response from an authorization/resource server.